source/ui/BioXAS/BioXASControlButton.cpp

```diff
@@ -0,0 +1,69 @@
+#include "BioXASControlButton.h"
+
+BioXASControlButton::BioXASControlButton(AMControl *control, QWidget *parent) :
+	AMToolButton(parent)
+{
+	// Initialize class variables.
+
+	control_ = 0;
+
+	greenValueSet_ = false;
+	greenValue_ = 0;
+
+	// Current settings.
+
+	setControl(control);
+
+	refresh();
```

Comment:
That call to `refresh()` isn't executed if the control given in the constructor is 0. For some widgets, this is important, as `refresh()` helps make sure the view is consistent with the current information. For this particular case, for this widget, you may be right--it might not make a difference.